server/src/main/java/org/opensearch/action/search/CreatePITRequest.java

```diff
@@ -0,0 +1,144 @@
+    @Nullable
+    private String preference = null;
+    private String[] indices = Strings.EMPTY_ARRAY;
+    private IndicesOptions indicesOptions = SearchRequest.DEFAULT_INDICES_OPTIONS;
+
+    public CreatePITRequest(TimeValue keepAlive, boolean allowPartialPitCreation) {
+        this.keepAlive = keepAlive;
+        this.allowPartialPitCreation = allowPartialPitCreation;
+    }
+
+    public CreatePITRequest(StreamInput in) throws IOException {
+        super(in);
+        indices = in.readStringArray();
+        indicesOptions = IndicesOptions.readIndicesOptions(in);
+        routing = in.readOptionalString();
+        preference = in.readOptionalString();
+        keepAlive = in.readTimeValue();
+        routing = in.readOptionalString();
+        allowPartialPitCreation = in.readBoolean();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeStringArray(indices);
+        indicesOptions.writeIndicesOptions(out);
+        out.writeOptionalString(preference);
+        out.writeTimeValue(keepAlive);
+        out.writeOptionalString(routing);
+        out.writeBoolean(allowPartialPitCreation);
+    }
+
+    public String getRouting() {
+        return routing;
+    }
+
+    public String getPreference() {
+        return preference;
+    }
+
+    public String[] getIndices() {
+        return indices;
+    }
+
+    public IndicesOptions getIndicesOptions() {
+        return indicesOptions;
+    }
+
+    public TimeValue getKeepAlive() {
+        return keepAlive;
+    }
+
+    public boolean isAllowPartialPitCreation() {
+        return allowPartialPitCreation;
+    }
+
+    public void setRouting(String routing) {
+        this.routing = routing;
+    }
+
+    public void setPreference(String preference) {
+        this.preference = preference;
+    }
+
+    public void setIndices(String[] indices) {
+        this.indices = indices;
+    }
+
+    public void setIndicesOptions(IndicesOptions indicesOptions) {
+        this.indicesOptions = Objects.requireNonNull(indicesOptions, "indicesOptions must not be null");
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+        if (keepAlive == null) {
+            validationException = addValidationError("Keep alive is missing", validationException);
+        }
+        return validationException;
+    }
+
+    @Override
+    public String[] indices() {
+        return indices;
+    }
+
+    @Override
+    public IndicesOptions indicesOptions() {
+        return indicesOptions;
+    }
+
+    public void setKeepAlive(TimeValue keepAlive) {
+        this.keepAlive = keepAlive;
+    }
+
+    @Override
+    public Task createTask(long id, String type, String action, TaskId parentTaskId, Map<String, String> headers) {
+        return new SearchTask(id, type, action, () -> "desc", parentTaskId, headers);
```

Comment:
Why does create PIT return a `SearchTask`

Reply:
we use search action as part of create pit flow which uses search task